modules/RTC/JitsiLocalTrack.js

```diff
@@ -265,8 +265,14 @@ export default class JitsiLocalTrack extends JitsiTrack {
      */
     _setRealDeviceIdFromDeviceList(devices) {
         const track = this.getTrack();
-        const device = devices.find(
-            d => d.kind === `${track.kind}input` && d.label === track.label);
+        const kind = `${track.kind}input`;
+        let device = devices.find(d => d.kind === kind && d.label === track.label);
+
+        if (!device && this._realDeviceId === 'default') { // the default device has been changed.
+            const label = (track.label || '').replace('Default - ', '');
+
+            device = devices.find(d => d.kind === kind && d.label === label);
+        }
```

Comment:
So what was the bug causing this issue? Maybe a comment here on why this replacing is needed would be good.

Reply:
If the default device was 'A' and for some reason the default device is changed to 'B' the label for the track will remain 'Default - A' but the label for the device in the device list will be 'A'.

I've put '// the default device has been changed.' comment. Do you want me to add the above explanation?

Reply:
Yes.

Reply:
Done!